### PR TITLE
Add UDP module

### PR DESCRIPTION
**Pull request by chokobole**
libtuv-DCO-1.0-Signed-off-by: wonyong.kim wonyong.kim@samsung.com

### CODE REVIEW

File: include/uv__udp.h [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+  UV_UDP_SEND_PRIVATE_FIELDS
+};
+
+int uv_udp_init(uv_loop_t*, uv_udp_t* handle);
+int uv_udp_init_ex(uv_loop_t*, uv_udp_t* handle, unsigned int flags);
+int uv_udp_open(uv_udp_t* handle, uv_os_sock_t sock);
+int uv_udp_bind(uv_udp_t* handle,
+                          const struct sockaddr* addr,
+                          unsigned int flags);
+
```
Comment (nova0821) [inline include/uv__udp.h]: Please fix the coding style. We have the rule "Return type should be on the same line as function name and parameters if it fits. If not, break between them aligned with the first argument".

File: include/uv__udp.h [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+int uv_udp_getsockname(const uv_udp_t* handle,
+                                 struct sockaddr* name,
+                                 int* namelen);
```
Comment (nova0821) [inline include/uv__udp.h]: ditto.

File: include/uv__udp.h [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+int uv_udp_send(uv_udp_send_t* req,
+                          uv_udp_t* handle,
+                          const uv_buf_t bufs[],
+                          unsigned int nbufs,
+                          const struct sockaddr* addr,
+                          uv_udp_send_cb send_cb);
```
Comment (nova0821) [inline include/uv__udp.h]: ditto.

File: include/uv__udp.h [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+int uv_udp_try_send(uv_udp_t* handle,
+                              const uv_buf_t bufs[],
+                              unsigned int nbufs,
+                              const struct sockaddr* addr);
```
Comment (nova0821) [inline include/uv__udp.h]: ditto.

File: include/uv__udp.h [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+int uv_udp_recv_start(uv_udp_t* handle,
+                                uv_alloc_cb alloc_cb,
+                                uv_udp_recv_cb recv_cb);
```
Comment (nova0821) [inline include/uv__udp.h]: ditto.

File: source/uv_udp.c [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+  if (handle->type != UV_UDP)
+    return UV_EINVAL;
+
+  if (addr->sa_family == AF_INET)
+    addrlen = sizeof(struct sockaddr_in);
+  else if (addr->sa_family == AF_INET6)
+    addrlen = sizeof(struct sockaddr_in6);
+  else
+    return UV_EINVAL;
+
```
Comment (nova0821) [inline source/uv_udp.c]: Please fix the coding style. Use brace if it has else part.

File: source/uv_udp.c [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+  if (handle->type != UV_UDP)
+    return UV_EINVAL;
+
+  if (addr->sa_family == AF_INET)
+    addrlen = sizeof(struct sockaddr_in);
+  else if (addr->sa_family == AF_INET6)
+    addrlen = sizeof(struct sockaddr_in6);
+  else
+    return UV_EINVAL;
```
Comment (nova0821) [inline source/uv_udp.c]: ditto.

File: source/uv_udp.c [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+}
+
+
+int uv_udp_recv_start(uv_udp_t* handle,
+                      uv_alloc_cb alloc_cb,
+                      uv_udp_recv_cb recv_cb) {
+  if (handle->type != UV_UDP || alloc_cb == NULL || recv_cb == NULL)
+    return UV_EINVAL;
+  else
+    return uv__udp_recv_start(handle, alloc_cb, recv_cb);
```
Comment (nova0821) [inline source/uv_udp.c]: ditto.

File: source/uv_udp.c [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+}
+
+
+int uv_udp_recv_stop(uv_udp_t* handle) {
+  if (handle->type != UV_UDP)
+    return UV_EINVAL;
+  else
+    return uv__udp_recv_stop(handle);
```
Comment (nova0821) [inline source/uv_udp.c]: ditto.

**Comment by chokobole** [comment]
@nova0821  Thanks for the review, I fixed as you had said!

File: source/unix/uv_unix_udp.c [hunk 10]
```diff
@@ -0,0 +1,636 @@
...
+  assert(handle->send_queue_count == 0);
+
+  /* Now tear down the handle. */
+  handle->recv_cb = NULL;
+  handle->alloc_cb = NULL;
+  /* but _do not_ touch close_cb */
+}
+
+
+
```
Comment (LaszloLango) [inline source/unix/uv_unix_udp.c]: minor style issue: too many empty lines

File: source/unix/uv_unix_udp.c [hunk 11]
```diff
@@ -0,0 +1,636 @@
...
+                   val,
+                   size);
+  }
+  if (r)
+    return -errno;
+
+  return 0;
+}
+
+static int uv__setsockopt_maybe_char(uv_udp_t* handle,
```
Comment (LaszloLango) [inline source/unix/uv_unix_udp.c]: missing new line (please use 2 new lines between functions)

**Comment by chokobole** [comment]
@LaszloLango Thanks for the review, I fixed as you had said!

**Comment by LaszloLango** [comment]
Thanks for the update. LGTM

**Pull request merged**